Sources/SwiftCuckoo/Lap.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+
+/// A model representing an individual lap within a session.
+///
+/// The `Lap` structure tracks the start time, end time, and calculates the duration
+/// of a single lap, providing a way to monitor intervals within a task or session.
+public struct Lap: Equatable, Hashable {
+
+    /// Defines the different states of a  `Lap`.
+    public enum LapStatus: Equatable {
+        case active                /// Indicates the lap is active.
+        case inactive              /// Indicates the lap is inactive.
+    }
+
+    /// A unique identifier for the lap instance.
+    public var id: Identifier
```

Comment:
Another important topic is to avoid it being public var, I don't remember if we did that on Session, but I believe the best thing here is to make only readable.

```suggestion
    public private(set) var id: Identifier
```

Reply:
Nice, just read up on private(set). I don't believe it is that way on Session.